translate/storage/base.py

```diff
@@ -759,7 +759,9 @@ def detect_encoding(self, text, default_encodings=None):
             if detected_encoding['confidence'] < 0.48:
                 detected_encoding = None
             elif detected_encoding['encoding'] == 'ascii':
-                detected_encoding['encoding'] = 'utf-8'
+                detected_encoding['encoding'] = self.encoding
+            else:
+                detected_encoding['encoding'] = detected_encoding['encoding'].lower()
         except ImportError:
             detected_encoding = None
```

Comment:
Dne Wed, 02 Sep 2015 08:38:22 -0700
Claude Paroz <notifications@github.com> napsal(a):

> > @@ -759,7 +759,9 @@ def detect_encoding(self, text, default_encodings=None):
> >              if detected_encoding['confidence'] < 0.48:
> >                  detected_encoding = None
> >              elif detected_encoding['encoding'] == 'ascii':
> > -                detected_encoding['encoding'] = 'utf-8'
> > +                detected_encoding['encoding'] = self.encoding
> > +            else:
> > +                detected_encoding['encoding'] = detected_encoding['encoding'].lower()
> >          except ImportError:
> >              detected_encoding = None
> 
> Could we add here:
> ```if text.startswith(codecs.BOM_UTF8):
>     detected_encoding = 'utf-8'
> if text.startswith(codecs.BOM_UTF16):
>     detected_encoding = 'utf-16'
> ```
> so that the test will pass even without chardet installed.

Added something like that in 3ed9e62.

-- 
	Michal Čihař | http://cihar.com | http://blog.cihar.com